ts/main.ts

```diff
@@ -132,3 +138,270 @@ async function renderAllCards(): Promise<void> {
+  switch (subclass) {
+    case 'Berserker':
+      return 'Barbarian Path of the Berserker';
+    case 'Champion':
+      return 'Fighter Champion Archetype';
+    case 'Devotion':
+      return 'Paladin Oath of Devotion';
+    case 'Draconic':
+      return 'Sorcerer Draconic Bloodline';
+    case 'Evocation':
+      return 'Wizard School of Evocation';
+    case 'Fiend':
+      return 'Warlock Fiend Patron';
+    case 'Hunter':
+      return 'Ranger Hunter Archetype';
+    case 'Land':
+      return 'Druid Circle of the Land';
+    case 'Life':
+      return 'Cleric Life Domain';
+    case 'Lore':
+      return 'Bard College of Lore';
+    case 'Open Hand':
+      return 'Monk Way of the Open Hand';
+    case 'Thief':
+      return 'Rogue Thief Archetype';
+    default:
+      return '';
+  }
+}
+
+let spellDetails: SpellDetails;
+
+$spellsListCardsDiv.addEventListener('click', async (event: Event) => {
+  try {
+    const $target = event.target as HTMLElement;
+
+    if (!$target.closest('div.card')) {
+      console.log('not a card');
+      return;
+    }
+
+    const $targetCard = $target.closest('div.card') as HTMLDivElement;
+    const cardSpellUrl = $targetCard.getAttribute('data-url');
+
+    $spellsListView.className += ' hidden';
+
+    if (!cardSpellUrl) throw new Error('cardSpellUrl does not exist');
+
+    await getSpellDetails(cardSpellUrl);
+
+    // NAME
+    $spellDetailsName.textContent = spellDetails.name;
+
+    // LEVEL, SCHOOL
+    $spellDetailsLevelSchool.textContent = levelNumberToString(
+      spellDetails.level,
+    );
+    if (spellDetails.level !== 0) {
+      $spellDetailsLevelSchool.textContent += ' Level ';
+    } else {
+      $spellDetailsLevelSchool.textContent += ' ';
+    }
+    $spellDetailsLevelSchool.textContent += spellDetails.school.name;
+
+    // CAST TIME
+    $spellDetailsCastTime.textContent = spellDetails.casting_time;
+
+    // RANGE
+    $spellDetailsRange.textContent = spellDetails.range;
+
+    // COMPONENTS
+    if (spellDetails.components.length === 1) {
+      $spellDetailsComponents.textContent = spellDetails.components[0];
+    } else {
+      for (let i = 0; i < spellDetails.components.length; i++) {
+        if (i === 0) {
+          $spellDetailsComponents.textContent =
+            spellDetails.components[i] + ', ';
+        } else if (i < spellDetails.components.length - 1) {
+          $spellDetailsComponents.textContent +=
+            spellDetails.components[i] + ', ';
+        } else {
+          $spellDetailsComponents.textContent += spellDetails.components[i];
+        }
+      }
+    }
+
+    // DURATION
+    $spellDetailsDuration.textContent = spellDetails.duration;
+
+    // CLEAR DESCRIPTIONS
+    while ($spellDetailsDescriptionDiv.childNodes.length > 0) {
+      if (!$spellDetailsDescriptionDiv.firstElementChild) break;
+      $spellDetailsDescriptionDiv.removeChild(
+        $spellDetailsDescriptionDiv.firstElementChild,
+      );
+    }
```

Comment:
Does the data can return different amount of descriptions for different spells?

Reply:
Yes, the data for descriptions is essentially an array of paragraphs, so I rendered those as separate <p> elements so I could space them apart.